cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -238,21 +239,9 @@ inline void VisitRawValuesInline(const ArrayType& values,
                                  VisitorNotNull&& visitor_not_null,
                                  VisitorNull&& visitor_null) {
   const auto data = values.raw_values();
-  if (values.null_count() > 0) {
-    BitmapReader reader(values.null_bitmap_data(), values.offset(), values.length());
-    for (int64_t i = 0; i < values.length(); ++i) {
-      if (reader.IsSet()) {
-        visitor_not_null(data[i]);
-      } else {
-        visitor_null();
-      }
-      reader.Next();
-    }
-  } else {
-    for (int64_t i = 0; i < values.length(); ++i) {
-      visitor_not_null(data[i]);
-    }
-  }
+  VisitBitBlocksVoid(
+      values.null_bitmap(), values.offset(), values.length(),
+      [&](int64_t i) { visitor_not_null(data[i]); }, [&]() { visitor_null(); });
```

Comment:
Did this have any impact on the sort benchmark?

Reply:
A couple significant speedups, the rest is a wash:
```
61         ArraySortIndicesInt64Narrow/32768/10000/min_time:1.000       1.502 GiB/sec       3.265 GiB/sec   117.397        {'run_name': 'ArraySortIndicesInt64Narrow/32768/10000/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 68135, 'null_percent': 0.01}
19             ArraySortIndicesInt64Narrow/32768/0/min_time:1.000       1.754 GiB/sec       3.485 GiB/sec    98.636            {'run_name': 'ArraySortIndicesInt64Narrow/32768/0/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 108233, 'null_percent': 0.0}
31           ArraySortIndicesInt64Narrow/32768/100/min_time:1.000       1.506 GiB/sec       2.432 GiB/sec    61.514           {'run_name': 'ArraySortIndicesInt64Narrow/32768/100/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 69677, 'null_percent': 1.0}
71         ArraySortIndicesInt64Narrow/1048576/100/min_time:1.000       1.445 GiB/sec       2.058 GiB/sec    42.464          {'run_name': 'ArraySortIndicesInt64Narrow/1048576/100/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 2079, 'null_percent': 1.0}
70         ArraySortIndicesInt64Narrow/8388608/100/min_time:1.000       1.176 GiB/sec       1.507 GiB/sec    28.098           {'run_name': 'ArraySortIndicesInt64Narrow/8388608/100/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 230, 'null_percent': 1.0}
53     TableSortIndicesInt64Narrow/1048576/100/1/1/min_time:1.000  152.839m items/sec  194.723m items/sec    27.404                            {'run_name': 'TableSortIndicesInt64Narrow/1048576/100/1/1/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 188}
45             ArraySortIndicesInt64Narrow/32768/2/min_time:1.000     872.577 MiB/sec       1.059 GiB/sec    24.262            {'run_name': 'ArraySortIndicesInt64Narrow/32768/2/min_time:1.000', 'run_type': 'iteration', 'repetitions': 0, 'repetition_index': 0, 'threads': 1, 'iterations': 37845, 'null_percent': 50.0}
```